src/editor/Gui.cpp

```diff
@@ -410,6 +417,30 @@ void Gui::draw3dContextMenus() {
 					command->execute();
 					app->pushUndoCommand(command);
 				}
+				if (ImGui::MenuItem("Export BSP model", 0, false, !app->isLoading)) {
+					if (app->pickInfo.modelIdx)
+					{
+						Bsp* tmpMap = new Bsp(map->path);
+						tmpMap->modelCount = 1;
+						tmpMap->models[0] = map->models[app->pickInfo.modelIdx];
+						STRUCTCOUNT removed = tmpMap->remove_unused_model_structures();

```

Comment:
This should be called after deleting all the entities, so that the other BSP models are deleted.